internal/services/cdn/cdn_frontdoor_origin_group_resource.go

```diff
@@ -0,0 +1,381 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceCdnFrontDoorOriginGroup() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceCdnFrontDoorOriginGroupCreate,
+		Read:   resourceCdnFrontDoorOriginGroupRead,
+		Update: resourceCdnFrontDoorOriginGroupUpdate,
+		Delete: resourceCdnFrontDoorOriginGroupDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.FrontDoorOriginGroupID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FrontDoorOriginGroupName,
+			},
+
+			"cdn_frontdoor_profile_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FrontDoorProfileID,
+			},
+
+			"load_balancing": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				MaxItems: 1,
+
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"additional_latency_in_milliseconds": {
+							Type:         pluginsdk.TypeInt,
+							Optional:     true,
+							Default:      50,
+							ValidateFunc: validation.IntBetween(0, 1000),
+						},
+
+						"sample_size": {
+							Type:         pluginsdk.TypeInt,
+							Optional:     true,
+							Default:      4,
+							ValidateFunc: validation.IntBetween(0, 255),
+						},
+
+						"successful_samples_required": {
+							Type:         pluginsdk.TypeInt,
+							Optional:     true,
+							Default:      3,
+							ValidateFunc: validation.IntBetween(0, 255),
+						},
+					},
+				},
+			},
+
+			// Optional
+			"health_probe": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"protocol": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(cdn.ProbeProtocolHTTP),
+								string(cdn.ProbeProtocolHTTPS),
+							}, false),
+						},
+
+						"request_type": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(cdn.HealthProbeRequestTypeGET),
+								string(cdn.HealthProbeRequestTypeHEAD),
+							}, false),
+						},
```

Comment:
Tbh I'd argue this, Protocol and Path want to be Required since these values are always going to be specific to the application/host being targeted, so we may as well make the three required so users are aware which healthcheck's configured to alleviate any confusion?

Reply:
I do not believe the `path` should be required, because depending on what `request_type` you use the `path` may not be needed since front door gets the URL from the backend. `path` is really only helpful in the `GET` scenario and most people do not use the `GET` `request_type` because every time the front door health check issues a `GET` call to ping the edge servers that edge server cost increases because it has to return a message-body in the response to the `GET` call. That said I do agree that the `interval_in_seconds` should be required as well, since that can affect the load on the edge servers, to forcing the end users to ping this number I think is totally valid.